app/controllers/api/v1/accounts/microsoft/authorizations_controller.rb

```diff
@@ -0,0 +1,26 @@
+class Api::V1::Accounts::Microsoft::AuthorizationsController < Api::V1::Accounts::BaseController
+  include MicrosoftConcern
+  before_action :check_authorization
+
+  def create
+    email = params[:authorization][:email]
+    redirect_url = microsoft_client.auth_code.authorize_url({
+                                                              redirect_uri: "#{base_url}/microsoft/callback",
+                                                              scope: 'offline_access https://outlook.office.com/IMAP.AccessAsUser.All
+                                                            https://outlook.office.com/SMTP.Send openid',
+                                                              prompt: 'consent'
+                                                            })
+    if redirect_url
+      ::Redis::Alfred.setex(email, Current.account.id)
```

Comment:
let's also pass the expiry as 5 minutes here or shorter. No need to persist the value for 1 day